model/map/src/main/java/org/keycloak/models/map/storage/chm/ConcurrentHashMapKeycloakTransaction.java

```diff
@@ -207,16 +209,21 @@ private V getUpdated(V orig) {
 
     @Override
     public V create(V value) {
-        K key = value.getId();
+        String key = value.getId();
+        if (key == null) {
+            K newKey = keyConvertor.yieldNewUniqueKey();
+            key = keyConvertor.keyToString(newKey);
+            value = Serialization.from(value, key);
```

Comment:
Can this be solved by introducing `void setId(String sKey)` to AbstractEntity?

Reply:
FTR: In here we need always to make a copy of a value from the parameter, to ensure that the caller does not get the direct reference to the entered entity (which could be of a completely different type than `Map*EntityImpl`)